test/hummingbot/client/ui/test_interface_utils.py

```diff
@@ -19,9 +37,11 @@ def test_format_bytes(self):
     @patch("hummingbot.client.ui.interface_utils._sleep", new_callable=AsyncMock)
     def test_start_timer(self, mock_sleep):
         mock_timer = MagicMock()
-        mock_sleep.side_effect = [None, ExpectedException()]
-        with self.assertRaises(ExpectedException):
-            asyncio.get_event_loop().run_until_complete(start_timer(mock_timer))
+        self.return_queue.put_nowait(None)
+        mock_sleep.side_effect = self.return_until_empty_then_signal_done_and_release_loop
+        task = self.ev_loop.create_task(start_timer(mock_timer))
+        self.async_run_with_timeout(self.done_event.wait())
+        task.cancel()
```

Comment:
I think we can avoid using a function that decides wether to keep running or not based on a queue by just taking advantage of the `side_effect` functionality of mock objects. If the side_effect is configured with a list of elements, it will return them in order as it gets called. And if one of the elements is an exception, it will trigger it when its turn arrives.
In this case we could configure the side effect with a list of `None` and an instance of asyncio.CancelledError (that technique is used in this test test/hummingbot/connector/exchange/liquid/test_liquid_exchange.py)